### PR TITLE
Add mhook for windows

### PR DESCRIPTION
**Pull request by yamashi**

### CODE REVIEW

File: packages/m/mhook/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    set_urls("https://github.com/apriorit/mhook/archive/$(version).zip")
+
+    add_versions("2.5.1", "37e7d65422a770b0a28835b4fe557823278f7118ff4a399d7acf345c8db318e5")
+
+    add_patches("2.5.1", "https://github.com/apriorit/mhook/commit/5ccb00a9c89280bfff7ce595873a9923415172a7.patch",
+                        "56561718ccf05c8c42fff05e6531cfa525cf93e0c9fa3bd226e74ef19eae1a1f")
+
+    on_install("windows", function (package)
+        local configs = {}
+        import("package.tools.cmake").install(package, configs, {buildir = "build"})
```
Comment (waruqi) [inline packages/m/mhook/xmake.lua:15]: If you do not use the build directory `, {buildir = "build"}`, you can remove it.

File: packages/m/mhook/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package("mhook")
+
+    set_homepage("https://github.com/martona/mhook")
+    set_description("A Windows API hooking library ")
+
+    set_urls("https://github.com/apriorit/mhook/archive/$(version).zip")
```
Comment (waruqi) [inline packages/m/mhook/xmake.lua:6]: Please add another git url, we can select the given tag/branch from git url.

Comment (yamashi) [reply packages/m/mhook/xmake.lua:6]: Do you mean adding git@github.com:apriorit/mhook.git ?

File: packages/m/mhook/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    set_urls("https://github.com/apriorit/mhook/archive/$(version).zip",
+             "https://github.com/apriorit/mhook.git")
+
+    add_versions("2.5.1", "37e7d65422a770b0a28835b4fe557823278f7118ff4a399d7acf345c8db318e5")
+
+    add_patches("2.5.1", "https://github.com/apriorit/mhook/commit/5ccb00a9c89280bfff7ce595873a9923415172a7.patch",
+                        "56561718ccf05c8c42fff05e6531cfa525cf93e0c9fa3bd226e74ef19eae1a1f")
+
+    on_install("windows", function (package)
+        local configs = {}
```
Comment (waruqi) [inline packages/m/mhook/xmake.lua:15]: Please add mode/kind configs.

```lua
        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
```

**Comment by waruqi** [comment]
Thanks! 👍

**Pull request merged**